internal/core/labels_test.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMergeCustomLabels(t *testing.T) {
+	t.Run("merge success", func(t *testing.T) {
+		// --- Given ---
+		dst := map[string]string{"A": "1", "B": "2"}
+		src := map[string]string{"B": "X", "C": "3"}
+
+		// --- When ---
+		err := MergeCustomLabels(dst, src)
+
+		// --- Then ---
```

Comment:
suggestion: lets remove the comments, as they don't tell the reader anything more than the code itself.

My rule for comments, is they should tell the user something they might otherwise not understand.

move below.
```suggestion
		dst := map[string]string{"A": "1", "B": "2"}
		src := map[string]string{"B": "X", "C": "3"}

		err := MergeCustomLabels(dst, src)
```